### ISSUE TITLE
Incorrect Documentation on DataModules

### ISSUE THREAD

**Issue by JJGO** [PyTorchLightning/pytorch-lightning#3353]
The current documentation on Datamodules [link](https://pytorch-lightning.readthedocs.io/en/latest/datamodules.html) appears to be incorrect. 

The initial MNIST example in the "What is a DataModule" section places the transforms under `setup` and feed them as arguments to the dataset constructor. 

However, in the "LightningDataModule API" documentation, all the `{train,val,test}_dataloader` examples create a transform and feed it into a `DataLoader`. This is incorrect, since PyTorch's [DataLoader](https://pytorch.org/docs/stable/data.html#torch.utils.data.DataLoader) does not take any `transform` argument.

More generally, it seems that the `{train,val,test}_dataloader` API examples suggest that LightingDataModules can deal with different transformations for all these three sets but there isn't a correct example for doing so and making it work that way seems a bit cumbersome with the current torchvision `VisionDataset` API that takes transforms on Dataset creation.

**Comment by awaelchli** [comment]
I agree that in the case of the dataloader, the transform is misplaced. The MNIST dataset however takes transforms, that should be correct, yes? 
Would you like to send a PR with suggested doc fixes? 
I'm removing the bug label, as I understand it's just documentation.

**Comment by JJGO** [comment]
I could send a PR fixing the `{train,val,test}_dataloader` examples but then they would be trivial (i.e. just calling and returning the `DataLoader`. 

However, the current docs seem to have a goal in mind to "decouple your data from transforms" (quoting directly from the docs). So ideally, the docs should provide examples on how to do that. This seems not trivial from the current `LightningDataModule` API. A potential fix would be to construct separate MNIST datasets for train, val and test and carefully `Subset` the train and val to ensure there is no common intersection. 

However, I was thinking whether some part of that logic could be offloaded to PyTorch Lightning itself. 

Let me know your thoughts
